Samples/auth/Outlook-Add-in-SSO-events/public/commands/commands.js

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+
+const urlOrigin = 'https://localhost:3000'; // Change this if deploying to a different location.
+
+Office.onReady((info) => {
+    if (info.host === Office.HostType.Outlook) {
+        // All console logs from events go to a runtime logging file. For more information, see https://learn.microsoft.com/office/dev/add-ins/testing/runtime-logging
+        console.log('initializing ...' + info);
+    }
+});
+
+// Default SSO settings for acquiring access tokens.
+const defaultSSO = {
+    allowSignInPrompt: false,
+    allowConsentPrompt: false,
+    //    forMSGraphAccess: true, // Leave commented during development testing (sideload) or you get a 13012 error from getAccessToken.
+};
+
+/**
+ * Handle the OnNewMessageCompose or OnNewAppointmentOrganizer event by calling getUserProfile.
+ * This appends a signature with the user's profile to the message body on send.
+ *
+ * @param {Office.AddinCommands.Event} event The OnNewMessageCompose or OnNewAppointmentOrganizer event object.
+ */
+function onItemComposeHandler(event) {
+    getUserProfile();
```

Comment:
This method should take the `event` object and call `event.completed()` after all logic is complete, otherwise, you are basically in state where you informed Outlook that you are done with all your logic while async calls are still running.

Reply:
okay, fixed.